CorsixTH/Lua/world.lua

```diff
@@ -1863,9 +1863,14 @@ function World:findFreeObjectNearToUse(humanoid, object_type_name, which, curren
   return object, ox, oy
 end
 
+--! Finds the most desirable room of a given room type
+--!param humanoid The queried NPC (start point)
+--!param room_type_id (string) The room we are looking for
```

Comment:
Code says it can also be `nil` is that something we want to keep?

Either way it's useful if the docs and the code agree on allowed input :)

Reply:
It does? The uses in the code don't seem to point to it ever using nil.
Do you mean the inference by `(not room_type_id or r.room_info.id == room_type_id)` ?

Reply:
Yes, that's what I mean.

If it unwanted, I'd suggest to assert that the parameter is  a string.

Reply:
I'll do some sanity runs and verify before updating the pr

Reply:
Otherwise you'd end up with any nearest room, which feels much less useful :)